### ISSUE TITLE
Encoding error when using convert_coco

### ISSUE THREAD

**Issue by oleg-pereziabov** [ultralytics/ultralytics#18390]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

When using ultralytics.data.converter.convert_coco in cases when image names in annotation file contain non-utf characters (e.g. cyrillic) there are encoding mistakes (not causing any errors, though) in created .txt files with labels. Haven't checked, but this may lead to errors on training process.
Seems, the case is here:
https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/converter.py#L269
So my solution to fix this is just to add encoding argument to context manager, like:
```python 
with open(json_file, encoding="utf-8") as f:
    data = json.load(f)
```
instead of
```python 
with open(json_file) as f:
    data = json.load(f)
```

### Environment

Ultralytics 8.3.49 🚀 Python-3.10.11 torch-2.4.1+cu121 CUDA:0 OS Windows-10 Install pip

### Minimal Reproducible Example

```python 
from ultralytics.data.converter import convert_coco

if __name__ == "__main__:
    path = "my_path_to_annotations_dir"
    save_path = "my_path_to_save_dir"
    convert_coco(labels_dir=path,
            use_segments=False,
            save_dir=save_path)
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @oleg-pereziabov, thank you for your interest in Ultralytics 🚀 and for bringing this to our attention!

We appreciate the detailed bug report and proposed solution! If this is indeed a 🐛 Bug Report, we recommend providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to ensure the issue can be effectively debugged. From your description and example, it seems you’ve already included relevant code—thanks! For further investigations, please confirm the steps to reproduce the behavior and provide any specific dataset or annotation snippet (if possible) showing the encoding issue.

In the meantime, please ensure you’re using the latest version of the `ultralytics` package, as updates may resolve this issue:

```bash
pip install -U ultralytics
```

### Resources

For additional assistance, check out the [Docs](https://docs.ultralytics.com), specifically the [minimum reproducible example guidelines](https://docs.ultralytics.com/help/minimum_reproducible_example/). Also, visit our [Discussions](https://github.com/ultralytics/ultralytics/discussions) or join the community on [Discord](https://discord.com/invite/ultralytics) 🎧, [Discourse](https://community.ultralytics.com), or [Reddit](https://reddit.com/r/Ultralytics) for more support.

### Verified Environments

YOLO can be run in any of the verified environments below. If you're not already, you might try running the script in one of these setups to check for consistency:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Providers**: [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), and [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

### Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across multiple environments.

Rest assured, an Ultralytics engineer will review your report and assist you further soon! Thank you for your willingness to contribute and for considering submitting a PR ❤️. Together, we can improve Ultralytics for everyone 🙂.

**Comment by Y-T-G** [comment]
Can you open a PR?

**Comment by oleg-pereziabov** [comment]
Sure, please check https://github.com/ultralytics/ultralytics/pull/18412

**Comment by glenn-jocher** [comment]
@oleg-pereziabov thank you for creating the PR! The team will review it soon. If you have additional suggestions or updates, feel free to share them directly in the PR.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has now been merged in [PR #18412](https://github.com/ultralytics/ultralytics/pull/18412)—kudos to the efforts of @oleg-pereziabov with contributions from @glenn-jocher! 🎉

### 🔧 Key Changes
This update enhances compatibility in the `convert_coco` function by ensuring JSON files are opened with explicit `UTF-8` encoding. This improvement addresses potential issues with non-ASCII characters found in datasets, making the tool more robust and accommodating for diverse global datasets.

### 🚀 How to Test
To confirm if this fix resolves the issue for you, you can either:
- Install the latest changes directly with `pip install git+https://github.com/ultralytics/ultralytics.git@main`, or  
- Wait for the next official release on PyPI.

Feedback is greatly appreciated to validate if this resolves the problem on your end! If the issue persists or if there are any additional concerns, please feel free to share further details.

Thanks again for reporting this issue—we’re always working to improve, and your input is invaluable. 🙏
